php/src/Google/Protobuf/Internal/Message.php

```diff
@@ -66,19 +66,30 @@ class Message
     /**
      * @ignore
      */
-    public function __construct($desc = NULL)
+    public function __construct($data = NULL)
     {
         // MapEntry message is shared by all types of map fields, whose
         // descriptors are different from each other. Thus, we cannot find a
         // specific descriptor from the descriptor pool.
-        if (get_class($this) === 'Google\Protobuf\Internal\MapEntry') {
-            $this->desc = $desc;
-            foreach ($desc->getField() as $field) {
-                $setter = $field->getSetter();
-                $this->$setter($this->defaultValue($field));
+        if ($data instanceof Descriptor) {
```

Comment:
I think you should check ```$this instanceof MapEntry```

Reply:
We can do this, but it seemed better to make this not hardcoded for a specific class, don't you think?

Reply:
The original behavior was only do the work for MapEntry if arg is descriptor. Your refactoring change the behavior.